ex04.ex

```diff
@@ -75,7 +76,11 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def reduce_even_odd([h|t], {state1,state2}, func), do: reduce_even_odd(t, func.(h, {state1,state2}), func)
```

Comment:
# -2

Wow: that took me a while to read!

You could have used regular `reduce` for this. I think that makes it simpler:

~~~ elixir
  def even_odd(list) do
    list
    |> reverse()
    |> reduce({ [], [] }, &partition_even_odd/2)
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}
~~~